src/Hive/Services/Common/ModService.cs

```diff
@@ -271,6 +275,66 @@ public async Task<HiveObjectQuery<Mod>> MoveMod(User? user, string channelDestin
             return new HiveObjectQuery<Mod>(StatusCodes.Status200OK, databaseMod);
         }
 
+        /// <summary>
+        /// Updates an existing mod.
+        /// </summary>
+        /// <param name="user">The user updating the mod.</param>
+        /// <param name="update">The data to update the mod with, as well as the mod identifier.</param>
+        /// <returns>A wrapped <see cref="Mod"/> of the updated mod, if successful.</returns>
+        public async Task<HiveObjectQuery<Mod>> UpdateMod(User? user, SerializedModUpdate update)
+        {
+            log.Debug("Getting database objects...");
+
+            if (update is null)
+                return new HiveObjectQuery<Mod>(StatusCodes.Status400BadRequest, "Mod Update is invalid");
+
+            var targetVersion = new Version(update.Version);
+
+            // Get the database mod that represents the SerializedModUpdate.
+            var databaseMod = await CreateModQuery().Where(x => x.ReadableID == update.ID && x.Version == targetVersion)
+                .FirstOrDefaultAsync().ConfigureAwait(false);
+
+            // The POSTed mod was successfully deserialzed, but no Mod exists in the database.
+            if (databaseMod == null)
```

Comment:
But: is null

Reply:
Nit*

Reply:
I copy pasted from the [MoveMod method](https://github.com/Atlas-Rhythm/Hive/blob/20d99577074f6107c411210cfea9ba793f38e65d/src/Hive/Services/Common/ModService.cs#L241) haha